src/global/global_helpers.R

```diff
@@ -6355,3 +6409,81 @@ ms_write_confdata <- function(x, which_dataset, to_where){
         stop('to_where must be either "local" or "remote"')
     }
 }
+
+rain_gauge_from_site_data <- function(network, domain, prodname_ms) {
+    
+    locations <- site_data %>%
+        filter(network == !!network,
+               domain == !!domain,
+               site_type == 'rain_gauge') 
+    
+    crs <- unique(locations$CRS) 
+    
+    if(length(crs) > 1) {
+        stop('crs is not consistent for all sites, cannot convert location in 
+             site_data to rain_gauge location product')
+    }
+    
+    locations <- locations %>%
+        sf::st_as_sf(coords = c('longitude', 'latitude'), crs = crs) %>%
+        select(site_name)
+    
+    path <- glue('data/{n}/{d}/derived/{p}',
+                 n = network,
+                 d = domain,
+                 p = prodname_ms)
+    
+    dir.create(path, recursive = TRUE)
+    
+    for(i in 1:nrow(locations)) {
+        
+        site_name <- pull(locations[i,], site_name)
+        
+        sf::st_write(locations[i,], glue('{p}/{s}',
+                                         p = path,
+                                         s = site_name),
+                     driver = 'ESRI Shapefile',
+                     delete_dsn = TRUE)
+    }
+}
+
+calc_inst_flux_wrap <- function(chemprod, qprod, prodname_ms) {
```

Comment:
Ah! I wish I'd checked this pull request before finishing my workday instead of after. We both decided to solve this problem ^_^